Source/CarthageKit/Project.swift

```diff
@@ -728,6 +756,47 @@ private func dSYMForFramework(frameworkURL: NSURL, inDirectoryURL directoryURL:
 		|> take(1)
 }
 
+/// Sends the URL to each bcsymbolmap found in the given directory.
+private func BCSymbolMapsInDirectory(directoryURL: NSURL) -> SignalProducer<NSURL, CarthageError> {
+	return filesInDirectory(directoryURL, fileExtension: "bcsymbolmap")
+}
+
+/// Sends the URLs of the bcsymbolmap files that match the given framework and are
+/// located somewhere within the given directory.
+private func BCSymbolMapsForFramework(frameworkURL: NSURL, inDirectoryURL directoryURL: NSURL) -> SignalProducer<NSURL, CarthageError> {
+	return UUIDsForFramework(frameworkURL)
+		|> flatMap(.Merge) { UUIDs in
+			if UUIDs.isEmpty {
+				return SignalProducer.empty
+			}
+			// conceptually, we just want to filter the URLs to find any that are in uuids.
+			// but we'd like to stop enumeration early when we find all the files.
+			// this also guarantees we don't send two URLs for bcsymbolmap files with the same UUID.
+			return SignalProducer { observer, disposable in
+				var remainingUUIDs = UUIDs
+				BCSymbolMapsInDirectory(directoryURL)
+					|> startWithSignal { signal, signalDisposable in
+						disposable += signalDisposable
+						signal.observe(Signal.Observer { event in
+							switch event {
+							case let .Next(value):
+								if let basename = value.value.lastPathComponent?.stringByDeletingPathExtension, fileUUID = NSUUID(UUIDString: basename) {
+									if remainingUUIDs.remove(fileUUID) != nil {
+										observer.put(event)
+										if remainingUUIDs.isEmpty {
+											sendCompleted(observer)
```

Comment:
> Does it satisfy you?

If I understand the `SignalProducer` operators correctly, any `SignalProducer` returned from the transform given to `flatMap` will only ever be started a maximum of once, right? Since only `flatten` ever actually gets ahold of those `SignalProducers` and it starts each producer it gets only once? As long as the `SignalProducer` returned by the transform is started at most once, then that code is safe and would work. If there's any way for the `SignalProducer` to be started multiple times, then they'd end up sharing the same dictionary and would misbehave.

If that assumption is valid, then I'd still probably tweak what you have to just subtracting elements from a copy of the `UUIDs` Set instead of building a dictionary, as the only info we need to persist is whether a particular UUID is valid.

Can you confirm that my assumption about `SignalProducers` returned from the transform given to `flatMap` is valid?